core/trino-main/src/main/java/io/trino/server/remotetask/TaskInfoFetcher.java

```diff
@@ -251,15 +251,15 @@ synchronized void updateTaskInfo(TaskInfo newTaskInfo)
         TaskStatus newRemoteTaskStatus = newTaskInfo.getTaskStatus();
 
         if (!newRemoteTaskStatus.getTaskId().equals(taskId)) {
-            log.warn("Task ID mismatch on remote task status.  Member task ID is %s, but remote task ID is %s.  This will confuse finalTaskInfo listeners.",
+            log.debug("Task ID mismatch on remote task status.  Member task ID is %s, but remote task ID is %s.  This will confuse finalTaskInfo listeners.",
```

Comment:
If these fill the logs, then I don't understand the implicit contract of this method.  These mismatched IDs can end up in `finalTaskInfo` here:
https://github.com/trinodb/trino/blob/f5def2df0ad9539351335a385db37d6db5c71614/core/trino-main/src/main/java/io/trino/server/remotetask/TaskInfoFetcher.java#L293
... and then any listeners added to this `TaskInfoFetcher` via [addFinalTaskInfoListener](https://github.com/trinodb/trino/blob/f5def2df0ad9539351335a385db37d6db5c71614/core/trino-main/src/main/java/io/trino/server/remotetask/TaskInfoFetcher.java#L165C37-L165C37) will be called with the task ID of some other task.  That other ID will be looked up in `StagePartition.tasks` here:
https://github.com/trinodb/trino/blob/f5def2df0ad9539351335a385db37d6db5c71614/core/trino-main/src/main/java/io/trino/execution/scheduler/faulttolerant/EventDrivenFaultTolerantQueryScheduler.java#L2683
... which results in action on some other task than where the listener was registered.  That can't be right, can it?

Are these warnings perhaps needlessly triggering when the task state is not `FINISHED`?  If so, we can add an additional guard to the `if` condition.

/cc @losipiuk

Reply:
I see lots of `Future calls to retrieveAndDropSpoolingOutputStats will fail`. And actually it makes sense as those are only set for FTE execution, and not for pipelined. 
Let's move those logs to `debug` and enable logging in configuration.
Separatly @dekimir  please send out PR to filter out this line for non FTE queries.

Reply:
OK.

And Martin is correct in saying this isn't actionable for Trino administrators, just developers.  Perhaps I should've used a `check` call instead of the log?

Reply:
checkargument would be fine here I think. We need to be sure that condition is fine - and will not backfire killing proper queries :)

Reply:
> please send out PR to filter out this line for non FTE queries.

https://github.com/trinodb/trino/pull/20216